### DISCUSSION TITLE
The interface protocol

### DISCUSSION THREAD

**Discussion by lvkas521424** [idaholab/moose#27558] [Q&A General]
Hello, I would like to package my program into the MOOSE framework. How can I find the interface protocol information of MOOSE and integrate it into MOOSE in a standardized manner? I have not found any relevant instructions on the official website yet.

yours,
lvkas

**Comment by GiudGiud** [comment]
Hello

The recommended way is to create an ExternalProblem class and set up the execution logic for your external program there. 

You can look to cardinal as an example of executing external programs within MOOSE
https://github.com/neams-th-coe/cardinal

Guillaume
